vortex-buffer/src/buffer_mut.rs

```diff
@@ -411,6 +421,50 @@ impl<T> FromIterator<T> for BufferMut<T> {
     }
 }
 
+impl Buf for ByteBufferMut {
+    fn remaining(&self) -> usize {
+        self.len()
+    }
+
+    fn chunk(&self) -> &[u8] {
+        self.as_slice()
+    }
+
+    fn advance(&mut self, cnt: usize) {
+        if !cnt.is_multiple_of(*self.alignment) {
+            vortex_panic!(
+                "Cannot advance buffer by {} items, resulting alignment is not {}",
+                cnt,
+                self.alignment
+            );
+        }
+        self.bytes.advance(cnt);
+        self.length -= cnt;
+    }
+}
+
+unsafe impl BufMut for ByteBufferMut {
+    fn remaining_mut(&self) -> usize {
+        self.bytes.remaining_mut()
```

Comment:
I don't think this is correct. The bytes package changed the meaning of BufMut to allow internally resizing: https://github.com/tokio-rs/bytes/issues/170

We should probably do the same. Annoyingly, this means this function returns a huge number, e.g. `usize::MAX - self.len()`.

Reply:
Ok, fixed